tests/test_subprocess_runner.py

```
@@ -592,3 +590,175 @@ def test_run_capture_timeout_returns_none_and_terminates():
+
+
+def test_bounded_probe_gives_up_instead_of_waiting(monkeypatch):
+    """A filesystem probe that never returns must not hold the caller.
+
+    Stands in for a `stat` wedged on an unresponsive mount, which cannot be
+    cancelled -- the thread is written off and the caller decides its own
+    fallback rather than the job hanging (issue #263).
+    """
+    monkeypatch.setattr(runner_module, "_STAT_TIMEOUT", 0.05)
+
+    async def _go():
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_go`

(ANN202)